Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/ruleset/RulesetManagement.java

```diff
@@ -321,5 +326,50 @@ public boolean isAlwaysShowingForKey(String keyId) {
         return false;
     }
 
+    @Override
+    public int updateMetadata(Collection<Metadata> currentMetadata, String acquisitionStage,
+            Collection<Metadata> updateMetadata) {
+
+        final Function<String, MutableTriple<Collection<Metadata>, Reimport, Collection<Metadata>>> entryProducer = key -> {
+            var entry = new MutableTriple<Collection<Metadata>, Reimport, Collection<Metadata>>();
+            entry.setLeft(new ArrayList<>());
+            entry.setRight(new ArrayList<>());
+            return entry;
+        };
+
+        HashMap<String, MutableTriple<Collection<Metadata>, Reimport, Collection<Metadata>>> unifying = new HashMap<>();
+        for (Metadata metadata : currentMetadata) {
+            unifying.computeIfAbsent(metadata.getKey(), entryProducer).getLeft().add(metadata);
+        }
+        for (Metadata metadata : updateMetadata) {
+            unifying.computeIfAbsent(metadata.getKey(), entryProducer).getRight().add(metadata);
+        }
+        Settings settings = ruleset.getSettings(acquisitionStage);
+        for (var entry : unifying.entrySet()) {
+            entry.getValue().setMiddle(settings.getReimport(entry.getKey()));
+        }
+
+        int sizeBefore = currentMetadata.size();
+        currentMetadata.clear();
+        for (var entry : unifying.entrySet()) {
+            var metadata = entry.getValue();
+            Collection<Metadata> currentEntries = metadata.getLeft();
+            Collection<Metadata> updateEntries = metadata.getRight();
 
+            switch (metadata.getMiddle()) {
+                case ADD:
+                    currentMetadata.addAll(currentEntries);
+                    currentMetadata.addAll(updateEntries);
+                    break;
+                case KEEP:
+                    currentMetadata.addAll(currentEntries);
+                    break;
+                case REPLACE:
+                    currentMetadata.addAll(updateEntries.isEmpty() ? currentEntries : updateEntries);
+                    break;
+                default: throw new IllegalStateException("complete switch");
```

Comment:
Please use an other error message here. If someone get this message as he use a non valid entry it would be very difficult to find the right place and the causing issue. Maybe something like that `"Used not supported reimport case {}"` where`{}` is replaced by the wrong value?